stdlib/Markdown/src/Markdown.jl

```diff
@@ -40,6 +40,18 @@ function docexpr(source::LineNumberNode, mod::Module, s, flavor = :julia)
     :($doc_str($(mdexpr(s, flavor)), $(QuoteNode(source)), $mod))
 end
 
+"""
+    macro md_str(s)
+
+Parse the given string as Markdown text and return a corresponding `Markdown.MD` object.
+
+# Example
+```jldoctest
+julia> html(md"# Hello, world!")
```

Comment:
Note that the `html` function also doesn't have a docstring…

Can we just show the example as
```jl
julia> md"# Hello, world!"
  Hello, world!
  ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡
```
similar to the way it appears in the REPL with `julia --color=no`?

Reply:
> Note that the `html` function also doesn't have a docstring 😢 …

I can give that a try as well (and generally have a look at the exported API). I'm wondering whether this would also warrant a more general documentation about how this can be used to export Markdown to various formats? I will have a general look.

> Can we just show the example as
> ```jl
> julia> s = md"# Hello, world!"
>   Hello, world!
>   ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡
> 
> julia> typeof(s)
> Markdown.MD
> ```
> similar to the way it appears in the REPL with `julia --color=no`? 

I tried this at some point but the bold formatting(?) gave me problems with the `doctest` failing after copy-and-pasting. I can try to see whether I can get it working after all. 

>  We could also add `show` calls, which *are* documented:
> ```jl
> julia> show(stdout, "text/markdown", s)
> # Hello, world!
> 
> julia> show(stdout, "text/html", s)
> <div class="markdown"><h1>Hello, world&#33;</h1>
> </div>
> 
> julia> show(stdout, "text/plain", s)
>   Hello, world!
>   ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡
> ```
> though it might be better to put those into a docstring for `Markdown.MD`.

This could be another option. I will play around a bit.

Thanks for your swift reply. I'll ping you again once I'm satisfied.

Reply:
Docstring for `html` added in #52733 ☺️

Reply:
I really like the example with `s = md"# Heading`, `typeof(s)`. Maybe we can just use this example without doctest to avoid the bold formatting problem?

Reply:
This now magically works on my machine, so I've reverted to `s = md"# Hello, world!"` and hope it won't break the build